src/wp-includes/class-wp-theme-json-resolver.php

```diff
@@ -145,19 +165,33 @@ protected static function translate( $theme_json, $domain = 'default' ) {
 	 * @return WP_Theme_JSON Entity that holds core data.
 	 */
 	public static function get_core_data() {
-		$config = static::read_json_file( __DIR__ . '/theme.json' );
-		$config = static::translate( $config );
+		if ( null !== static::$core && static::has_same_blocks( 'core' ) ) {
+			return static::$core;
+		}
 
-		/**
-		 * Filters the default data provided by WordPress for global styles & settings.
-		 *
-		 * @since 6.1.0
+		/*
+		 * We only need to do once the following processes:
 		 *
-		 * @param WP_Theme_JSON_Data Class to access and update the underlying data.
+		 * - read data from the filesystem
+		 * - translate
+		 * - run filters
 		 */
-		$theme_json   = apply_filters( 'theme_json_default', new WP_Theme_JSON_Data( $config, 'default' ) );
-		$config       = $theme_json->get_data();
-		static::$core = new WP_Theme_JSON( $config, 'default' );
+		static $core_data = null;
+		if ( null === $core_data ) {
+			$core_data = static::read_json_file( __DIR__ . '/theme.json' );
+			$core_data = static::translate( $core_data );
+			/**
+			 * Filters the default data provided by WordPress for global styles & settings.
+			 *
+			 * @since 6.1.0
+			 *
+			 * @param WP_Theme_JSON_Data Class to access and update the underlying data.
+			 */
+			$theme_json   = apply_filters( 'theme_json_default', new WP_Theme_JSON_Data( $core_data, 'default' ) );
```

Comment:
```suggestion
                         $theme_json = new WP_Theme_JSON_Data( $core_data, 'default' );
			/**
			 * Filters the default data provided by WordPress for global styles & settings.
			 *
			 * @since 6.1.0
			 *
			 * @param WP_Theme_JSON_Data Class to access and update the underlying data.
			 */
			$theme_json   = apply_filters( 'theme_json_default',  $theme_json );
```

Cleaner to break these into two lines for readability.